Lesson9

JuicyFlash opened the pull request:
- реализован ресурс answer
- немного изменены формы отображения информации
 По заданию:
- добавлена новая миграция с необходимыми атрибутами для развязочной таблицы processed_tests (correct_questions, question_number, current_question)
- реализован ресурс прохождения теста processed_test
- на форме прохождения теста отображается номер вопроса и общее количество вопросов теста
- на форме результатов водится сообщение и процент успеха в зависимости от количества правильных ответов + выделение цветом

JuicyFlash committed:
view show для ресурса question

JuicyFlash committed:
полная реализация ресурса answer

JuicyFlash committed:
реализован ресурс processed_test - прохождение теста

JuicyFlash committed:
добавлен номер ответа

JuicyFlash committed:
форма с результатами + выделение цветом

BubuntuClu commented on app/controllers/tests_controller.rb:
```diff
@@ -8,4 +8,11 @@ def show
   end
 
+  def start
+    @test = Test.find(params[:id])
```
этот код используется в двух экшенах, можно вынести в колбэк

BubuntuClu commented on app/models/processed_test.rb:
```diff
@@ -4,5 +4,43 @@ class ProcessedTest < ApplicationRecord
+      self.correct_questions += 1
+    end
+    self.question_number += 1 if test.questions.count < question_number
+    self.current_question = next_question
```
след вопрос надо назначать через колбэк, а не в методе

BubuntuClu commented on app/views/processed_tests/result.html.erb:
```diff
@@ -0,0 +1,7 @@
+<%if @processed_test.percent_result >= 85 %>
```
85 число непонятное без контекста, надо поместить в  константу в модели

так же само сравнение можно поместить в метод модели `success?`

JuicyFlash committed:
дублирующийся код убран в колбэк

JuicyFlash committed:
логика следующего вопроса перенесена из метода в колбэк
логика определения корректности прохождения теста полностью перенесена в модель

JuicyFlash commented:
- в контроллере TestsController перенёс дублирующийся код в колбэк
- следующий вопрос теперь устанавливается с помощью колбэка модели ProcessedTest
- результат прохождения теперь полностью определяется в модели ProcessedTest

BubuntuClu commented on app/models/processed_test.rb:
```diff
@@ -6,27 +6,38 @@ class ProcessedTest < ApplicationRecord
+    percent_result >= percents_for_pass_test
+  end
+
+  def percent_for_pass_test
```
это не константа. перенеси значение в константу

JuicyFlash committed:
константа
